modules/@apostrophecms/piece-type/index.js

```diff
@@ -1174,15 +1176,11 @@ module.exports = {
               };
               inserts.push(newDoc);
 
-              if (inserts.length > 100) {
-                self.flush(deletes, inserts);
-                deletes = [];
-                inserts = [];
-              }
+              await self.flushInsertsAndDeletes(inserts, deletes);
             }
           });
 
-          self.flush(deletes, inserts);
+          await self.flushInsertsAndDeletes(inserts, deletes);
```

Comment:
You need a third argument here to say "I don't care if there are 100 of them or not, we need to flush the rest now."